### DISCUSSION TITLE
How to set temperature for CompleteChatAsync options?

### DISCUSSION THREAD

**Discussion by kovac031** [openai/openai-dotnet#251] [Q&A] [answered]
I am sending a `List<ChatMessage> prompt` and a `ChatCompletionOptions options`, however from what I can see `ChatCompletionOptions` doesn't have any `Temperature` property. The goal is to reduce the creativity of the response. I can see in OpenAI API documentation that such a setting exists.

This is my current implementation, I've added `Temperature` and it doesn't break anything but I also am not quite sure if it does anything either:

    ChatCompletionOptions options = new()
    {
        ResponseFormat = ChatResponseFormat.CreateJsonSchemaFormat(
            jsonSchemaFormatName: "email_response",
            jsonSchema: BinaryData.FromBytes("""
            {
                "type": "object", 
                "properties": { 
                    "emailTitle": { "type": "string" },
                    "emailBody": { "type": "string" }
                },
                "required": ["emailTitle", "emailBody"],
                "additionalProperties": false
            }
            """u8.ToArray()),
            jsonSchemaIsStrict: true
        ),
        MaxOutputTokenCount = 1500, 
        Temperature = 0.1f
    };

How should I be setting temperature?

For comparison, here is a solution in JS I found [here](https://community.openai.com/t/questions-about-roles-and-chat-history/452170), it's structured differently, but is setting the temperature which I want:

    const result = await openai.chat.completions.create({
    messages: [
    { role: “system”, content: instructions },
    { role: “user”, content: message },
    { role: “assistant”, content: metadataString },
    ],
    model: “gpt-3.5-turbo”,
    max_tokens: 150,
    Temperature: 0.3,
    });

**Comment by joseharriaga** [comment, marked as answer]
Thank you for reaching out, @kovac031 ! `ChatCompletionOptions` does have a `Temperature` property, as you can see here:
🔗 https://github.com/openai/openai-dotnet/blob/main/api/OpenAI.netstandard2.0.cs#L1387
🔗 https://github.com/openai/openai-dotnet/blob/main/src/Generated/Models/ChatCompletionOptions.cs#L46

The code that you included above is the correct way to use it.

**Reply by kovac031** [reply]
I am inexperienced. I am unable to navigate to these files from within my project where I installed this library. I use Visual Studio.

Searching for partial class ChatCompletionOptions only ever returns one result, going to definition navigates only to the one I am showing in the attached screenshot.

![fsfsf324rwfs](https://github.com/user-attachments/assets/2b0b8c4a-1312-4152-b553-851ae9c5d066)

Would you be so kind as to explain why am I unable to open the other parts of the partial class from VS? Or do I simply don't know how?
